Sources/Fluent/Schema/Schema+Field.swift

```diff
@@ -11,14 +11,31 @@ extension Schema {
         public var `default`: Node?
 
         public enum DataType {
-            case id
+            case id(keyType: KeyType?)
             case int
             case string(length: Int?)
             case double
             case bool
             case data
             case custom(type: String)
         }
+        
+        public enum KeyType {
```

Comment:
Reason I proposed a type here is it'd be consistent with the `DataType` (and it is indeed a field related thing). I also wasn't sure if the type would get referenced also in other contexts beside the .id case of DataType (for example foreign key related uses, or cases where this type would be used to describe something that is not necessarily the primary key of the record, which I've understood .id refers to). No concrete issues come to mind though, happy to do the change.